google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -102,6 +117,44 @@ public BulkWriter bulkWriter(BulkWriterOptions options) {
     return new BulkWriter(this, options);
   }
 
+  @Nonnull
+  public ApiFuture<Void> recursiveDelete(CollectionReference reference) {
+    BulkWriter writer = getBulkWriter();
+    RecursiveDelete deleter = new RecursiveDelete(this, writer, reference);
+    return deleter.run();
+  }
+
+  @Nonnull
+  public ApiFuture<Void> recursiveDelete(CollectionReference reference, BulkWriter bulkWriter) {
+    RecursiveDelete deleter = new RecursiveDelete(this, bulkWriter, reference);
+    return deleter.run();
+  }
+
+  @Nonnull
+  public ApiFuture<Void> recursiveDelete(DocumentReference reference) {
+    BulkWriter writer = getBulkWriter();
+    RecursiveDelete deleter = new RecursiveDelete(this, writer, reference);
+    return deleter.run();
+  }
+
+  @Nonnull
+  public ApiFuture<Void> recursiveDelete(DocumentReference reference, BulkWriter bulkWriter) {
+    RecursiveDelete deleter = new RecursiveDelete(this, bulkWriter, reference);
+    return deleter.run();
+  }
+
+  /**
+   * This overload is only used to test the query resumption with startAfter() once the
+   * RecursiveDelete instance has MAX_PENDING_OPS pending.
+   */
+  @Nonnull
+  @VisibleForTesting
+  ApiFuture<Void> recursiveDelete(
+      CollectionReference reference, BulkWriter bulkWriter, int maxLimit, int minLimit) {
+    RecursiveDelete deleter = new RecursiveDelete(this, bulkWriter, reference, maxLimit, minLimit);
```

Comment:
This code flow would be a bit cleaner if you called this method from all overloads. This will ensue that all calls go through the same callsite. If you take `parentPath` and `collectionId` as input here then you will also remove some duplication in RecursiveDelete (as you only need one constructor).